includes/class-civicrm-caldera-forms-helper.php

```diff
@@ -484,13 +484,43 @@ public function map_fields_to_prerender( $config, &$form, $ignore_fields, $entit
 				if ( $field['type'] == 'radio' ) {
 					$options = Caldera_Forms_Field_Util::find_option_values( $field );
 					$form['fields'][$field['ID']]['config']['default'] = array_search( $value, $options );
+				} elseif ( $field['type'] == 'date_picker' ) {
+					// If the date is still in the format returned by the CiviCRM API, convert it to the date picker's format.
+					$filtered_value = &$form['fields'][$field['ID']]['config']['default'];
+					if ( preg_match('/^\d{4}-\d{2}-\d{2}(?: \d{2}:\d{2}:\d{2})?$/', $filtered_value ) ) {
+						$format = self::translate_date_picker_format( $field['config']['format'] );
+						$filtered_value = date_create( $filtered_value )->format( $format );
+					}
 				}
 			}
 		}
 
 		return $form;
 	}
 
+	/**
+	 * Translate Caldera Forms date picker formats to PHP date formats.
+	 *
+	 * @since 1.0.2
+	 *
+	 * @param string $date_picker_format The Caldera Forms date picker format
+	 */
+	public static function translate_date_picker_format( $date_picker_format ){
```

Comment:
No need for static here.

Reply:
There's no _need_ for it, but my understanding of PHP (and OOP in general) is that methods that do not operate on a specific object ($this) should be static. There may even be a slight benefit in terms of speed and memory usage. That said, it's your repo, so if you want me to remove the static I'll do it.

Reply:
I agree that in this case is a _class_ as opposed to an _object_ and yes it _could_ be `static` but for consistency sake throughout the codebase I prefer non statics here :) each class in this project has a `static` reference to the plugin and to this class' instance through `$this->plugin->helper->some_helper_method()`.

Reply:
OK, I will remove the static, but that means I'll have to change the way the way the function is called. It was called with `self::translate_date_picker_format()` but I'm going to change that to `$this->translate_date_picker_format()` because, as the PHP Manual [warns](https://www.php.net/manual/en/language.oop5.static.php):

> In PHP 7, calling non-static methods statically is deprecated, and will generate an E_DEPRECATED warning. Support for calling non-static methods statically may be removed in the future.

But this means that if the calling function `map_fields_to_prerender` is ever called statically, it will generate a fatal error "Using $this when not in object context".

Reply:
Thank you @kirk-circle, `map_fields_to_prerender` is not called statically throughout the code and it shouldn't be called statically as it's not a static method, sorry, I didn't explained myself properly, the helper class `CiviCRM_Caldera_Forms_Helper` instance and rest of instances sit on a property (ie `$helper`) on the main plugin's [class](https://github.com/mecachisenros/caldera-forms-civicrm/blob/master/caldera-forms-civicrm.php#L30) which is a singleton and each instance has a reference to that singleton.

Reply:
OK, well I think we're all done then. 🙂